// casual/src/providers/BY/address.js
var provider = {
	city_prefixes: ['г.'],

	region_suffixes: ['вобл.'],

	street_prefixes: [
		'пер.', 'вул.', 'пр.', 'шассе', 'пл.'
	],

	zip_formats: ['######'],

	countries: [
		'Рэспублiка Беларусь'
	],

	regions: [
		'Мiнская', 'Брэсцская', 'Гомельская', 'Гродненская', 'Вiцебская', 'Магiлёуская'
	],

	cities: [
		'Баранавiчы', 'Брэст', 'Вiцебск', 'Гомель', 'Гродна',
		'Бяроза', 'Слуцк', 'Смалевiчы', 'Пiнск', 'Слонiм', 'Лiда', 'Орша', 'Маладзечна',
		'Салiгорск', 'Полацк', 'Наваполацк'
	],

	streets: [
		'Ленiна', 'Савецкая', 'Айвазоуская', 'Бярэзiнская', 'Брэсцкая', 'Мiнская', '1905 года', 'Чэхава', 'Сталiна',
		'Косманаутау', 'Гагарына', 'Славы', 'Зоi Касмадзямъянскай', '50 лет ВЛКСМ'
	],

	address_formats: [
		"{{region}} {{region_suffix}}, {{city_prefix}} {{city}}, {{street_prefix}} {{street}}, {{building_number}}",
	],

	region_suffix: function() {
		return this.random_element(this.region_suffixes);
	},

	region: function() {
		return this.random_element(this.regions);
	},

	city: function() {
		return this.random_element(this.cities);
	},

	street_prefix: function() {
		return this.random_element(this.street_prefixes);
	},

	street: function() {
		return this.random_element(this.streets);
	}
}

module.exports = provider;
