// casual/src/providers/BY/person.js
var provider = {
	phone_formats: [
		'+375 29 ###-##-##',
	],

	full_name_formats: [
		'{{last_name_male}} {{first_name_male}} {{middle_name_male}}',
		'{{last_name_female}} {{first_name_female}} {{middle_name_female}}'
	],

	male_name_formats: [
		'{{first_name_male}} {{middle_name}} {{last_name}}',
		'{{last_name}} {{first_name_male}} {{middle_name}}'
	],

	female_name_formats: [
		'{{first_name_female}} {{middle_name}} {{last_name}}',
		'{{last_name}} {{first_name_female}} {{middle_name}}'
	],

	name_formats: [
		'{{first_name_male}} {{last_name_male}}',
		'{{first_name_female}} {{last_name_female}}',
		'{{last_name_male}} {{first_name_male}}',
		'{{last_name_female}} {{first_name_female}}'
	],

	first_name_males: [
		'Абрам', 'Аугуст', 'Адам', 'Адрыян', 'Акiм', 'Аляксандр', 'Аляксей', 'Альберт', 'Ананiй', 'Анатолiй', 'Андрэй', 'Антон', 'Антонiн',
		'Апалон', 'Аркадзiй', 'Арсенiй', 'Арцемiй', 'Артур', 'Арцём', 'Афанасiй', 'Багдан', 'Балеслау', 'Барыс', 'Банiслав', 'Вадзiм',
		'Валянцiн', 'Валерыян', 'Валерый', 'Васiлiй', 'Венiамiн', 'Вiкенцiй', 'Вiктар', 'Вiль', 'Вiталiй', 'Вiтальд', 'Улад', 'Уладзiмiр',
		'Уладзiслау', 'Герман', 'Глеб', 'Гардей',
		'Грыгорый', 'Давiд', 'Дан', 'Данiiл', 'Данiла', 'Дзянiс', 'Дмiтрiй', 'Дабрыня', 'Данат', 'Яугенiй', 'Ягор', 'Ефiм',
		'Захар', 'Iван', 'Iгнат', 'Iгнацiй', 'Iгар', 'Iлларыон', 'Iлья', 'Iммануiл', 'Iннакенцiй', 'Iосiф', 'Iраклiй', 'Кiрыл',
		'Клiм', 'Канстанцiн', 'Кузьма', 'Лаурэнцiй', 'Леу', 'Леанiд', 'Макар', 'Максiм', 'Марат', 'Марк', 'Матвей', 'Мiлан',
		'Мiраслау', 'Мiхаiл', 'Назар', 'Нестар', 'Нiкiта', 'Нiкадзiм', 'Нiкалай', 'Алег', 'Павел', 'Платон', 'Прохар', 'Пётр',
		'Радзiслау', 'Рафаiл', 'Роберт', 'Радзiон', 'Раман', 'Расцiслау', 'Руслан', 'Сямён', 'Сяргей',
		'Спартак', 'Станiслау', 'Сцяпан', 'Тарас', 'Цiмафей', 'Цiмур'
	],

	first_name_females: [
		'Аляксандра', 'Алiна', 'Алiса', 'Ала', 'Альбiна', 'Алёна', 'Анастасiя', 'Анжэлiка', 'Анна', 'Антанiна', 'Анфiса', 'Валянцiна',
		'Валерыя',
		'Варвара', 'Васялiса', 'Вера', 'Веранiка', 'Вiкторыя', 'Уладлена', 'Галiна', 'Даръя', 'Дзiана', 'Дзiна', 'Дамiнiка', 'Ева',
		'Яугенiя', 'Екацерына', 'Елена', 'Елiзавета', 'Жана', 'Зiнаiда', 'Злата', 'Зоя', 'Iзабелла', 'Iзольда', 'Iнга', 'Iнэса',
		'Iнна', 'Iрына', 'Iскра', 'Капiталiна', 'Клаудзiя', 'Клара', 'Клеменцiна', 'Крысцiна', 'Ксенiя', 'Лада', 'Ларыса', 'Лiдзiя',
		'Лiлiя', 'Любоу', 'Людмiла', 'Люся', 'Майя', 'Мальвiна', 'Маргарыта', 'Марына', 'Марыя', 'Марта', 'Надзежда', 'Наталья',
		'Неллi', 'Нiка', 'Нiна', 'Нонна', 'Оксана', 'Олеся', 'Ольга', 'Полiна', 'Рада', 'Раiса', 'Регiна', 'Рената',
		'Разалiна', 'Святлана', 'Софья', 'Сафiя', 'Таiсiя', 'Тамара', 'Таццяна', 'Ульяна', 'Фаiна', 'Федосья', 'Фларэнцiна', 'Эльвiра', 'Эмiлiя',
		'Эмма', 'Юлiя', 'Яна', 'Яраслава'
	],

	middle_name_males: [
		'Аляксандравiч', 'Аляксеевiч', 'Андрэевiч', 'Дзмiтрыевiч', 'Яугеньевiч',
		'Сяргеевiч', 'Iванавiч', 'Фёдаравiч', 'Львавiч', 'Раманавiч', 'Уладзiмiравiч',
		'Барысавiч', 'Максiмавiч'
	],

	middle_name_females: [
		'Аляксандрауна', 'Аляксееуна', 'Андрэеуна', 'Дзмiтрыеуна', 'Евгеньевна',
		'Сергеевна', 'Iванауна', 'Фёдоравна', 'Львавна', 'Романавна', 'Уладзiмiрауна',
		'Барiсауна', 'Максiмауна'
	],

	last_name_males: [
		'Смiрноу', 'Iваноу', 'Кузняцау', 'Соколау', 'Попау', 'Лебедев', 'Козлау',
		'Науiкау', 'Морозау', 'Петрау', 'Волкау', 'Солауьёв', 'Васiльев', 'Зайцев',
		'Павлау', 'Семёнау', 'Голубев', 'Вiноградау', 'Богданау', 'Воробьёв',
		'Фёдорау', 'Мiхайлау', 'Беляев', 'Тарасау', 'Белау', 'Комарау', 'Орлау',
		'Кiселёв', 'Макарау', 'Андрэеу', 'Кавалёу', 'Iльiн', 'Гусеу', 'Цiтоу',
		'Кузьмiн', 'Кудрауцау', 'Баранау', 'Кулiкоу', 'Аляксееу', 'Сцяпанау',
		'Якаулеу', 'Сарокiн', 'Сяргееу', 'Раманоу', 'Захарау', 'Барысау', 'Коралёу',
		'Герасiмау', 'Панамароу', 'Грыгорьеу', 'Лазарау', 'Мядзведзеу', 'Яршоу',
		'Нiкiцiн', 'Собалеу', 'Рабоу', 'Палякоу', 'Цветкоу', 'Данiлоу', 'Жукау',
		'Фролау', 'Журавлёв', 'Нiколаев', 'Крылау', 'Максiмау', 'Сiдорау', 'Осiпау',
		'Белоусау', 'Федотав', 'Дорофееу', 'Егорау', 'Матвеев', 'Баброу', 'Дзмiтрыеу',
		'Калiнiн', 'Анiсiмау', 'Петухоу', 'Антонау', 'Цiмофеев', 'Нiкiфорау',
		'Веселау', 'Фiлiпау', 'Маркау', 'Большакау', 'Суханау', 'Мiронау', 'Шiряев',
		'Александрау', 'Конауалау', 'Шестакау', 'Казакау', 'Ефiмау', 'Денiсау',
		'Громау', 'Фомiн', 'Давыдау', 'Мельнiкау', 'Щербакау', 'Блiнау', 'Колеснiкау',
		'Карпау', 'Афанасьев', 'Власау', 'Маслау', 'iсакау', 'Тiхонау', 'Аксёнау',
		'Гаврiлау', 'Родзiонау', 'Котау', 'Гарбуноу', 'Кудряшоу', 'Быкав', 'Зуев',
		'Третьякау', 'Савельев', 'Панау', 'Рыбакау', 'Суворау', 'Абрамау', 'Воронау',
		'Мухiн', 'Архiпау', 'Трофiмау', 'Мартынау', 'Емельянау', 'Горшкау', 'Чарноу',
		'Аучыннiкау', 'Селязнёу', 'Панфiлау', 'Копылау', 'Мiхеев', 'Галкiн', 'Назарау',
		'Лобанау', 'Лукiн', 'Белякоу', 'Патапау', 'Некрасау', 'Хахлоу', 'Жданав',
		'Наумау', 'Шiлау', 'Воронцау', 'Ермакау', 'Дроздау', 'iгнатьев', 'Савiн',
		'Логiнау', 'Сафонау', 'Капустiн', 'Кiрiллау', 'Моiсеев', 'Елiсеев', 'Кошелев',
		'Костiн', 'Горбачёв', 'Орехау', 'Ефремау', 'iсаев', 'Евдокiмау', 'Калашнiкау',
		'Кабанау', 'Носкау', 'Юдiн', 'Кулагiн', 'Лапiн', 'Прохорау', 'Нестерау',
		'Харытонау', 'Агафонау', 'Мурауьёу', 'Ларыонау', 'Федасеев', 'Зiмiн', 'Пахомау',
		'Шубiн', 'Iгнатав', 'Фiлатау', 'Крюкау', 'Рогау', 'Кулакау', 'Цярэнцьеу',
		'Малчанау', 'Уладзiмiрау', 'Арцемьеу', 'Гурьеу', 'Зiнауьеу', 'Грышiн', 'Кононау',
		'Дзяменцьеу', 'Сiтнiкау', 'Сiманау', 'Мiшын', 'Фадзеев', 'Камiсарау', 'Мамонтау',
		'Носау', 'Гуляев', 'Шарау', 'Устiнау', 'Вiшнякау', 'Яусеев', 'Лаврентьеу',
		'Брагiн', 'Константiнау', 'Корнiлау', 'Авдеев', 'Зыкау', 'Бiрюкау', 'Шарапау',
		'Нiконау', 'Щукiн', 'Дьячкау', 'Одiнцау', 'Сазонау', 'Якушев', 'Красiльнiкау',
		'Гардееу', 'Самойлау', 'Князеу', 'Беспалау', 'Уварау', 'Шашкоу', 'Бобылёв',
		'Доронiн', 'Белозёрау', 'Рожкау', 'Самсонау', 'Мяснiкау', 'Лiхачёв', 'Бурау',
		'Сысоев', 'Фамiчав', 'Русакоу', 'Стралкоу', 'Гушчын', 'Цяцерын', 'Колабау',
		'Субоцiн', 'Фокiн', 'Блохiн', 'Селiверстау', 'Пестоу', 'Кондраццеу', 'Сiлiн',
		'Меркушоу', 'Лыткiн', 'Турау'
	],

	last_name_females: [
		'Смiрнав', 'Iванав', 'Кузняцав', 'Сакалав', 'Папав', 'Лебедзев', 'Казлав',
		'Навiкав', 'Марозав', 'Пятрав', 'Волкав', 'Салауёв', 'Васiльев', 'Зайцев',
		'Паулав', 'Сямёнав', 'Голубев', 'Вiноградав', 'Багданав', 'Варабьёв',
		'Фёдарав', 'Мiхайлав', 'Беляев', 'Тарасав', 'Белав', 'Камарав', 'Арлав',
		'Кiсялёв', 'Макарав', 'Андрэев', 'Кавалёв', 'Iльiн', 'Гусев', 'Цiтав',
		'Кузьмiн', 'Кудрауцэв', 'Баранав', 'Кулiкав', 'Аляксеев', 'Сцяпанав',
		'Якаулев', 'Сарокiн', 'Сяргеев', 'Раманав', 'Захарав', 'Борiсав', 'Королёв',
		'Герасiмав', 'Пономарёв', 'Грыгорьев', 'Лазарев', 'Мядзведзев', 'Яршав',
		'Нiкiтiн', 'Собалев', 'Рабав', 'Палякав', 'Цвяткав', 'Данiлав', 'Жукав',
		'Фролав', 'Журавлёв', 'Нiколаев', 'Крылав', 'Максiмав', 'Сiдорав', 'Осiпав',
		'Белоусав', 'Федотав', 'Дорофеев', 'Егорав', 'Матвеев', 'Бобрав', 'Дзмiтрыев',
		'Калiнiн', 'Анiсiмав', 'Петухав', 'Антонав', 'Цiмофеев', 'Нiкiфарав',
		'Вяселав', 'Фiлiпав', 'Маркав', 'Большакав', 'Суханав', 'Мiронав', 'Шыряев',
		'Александрав', 'Конавалав', 'Шестакав', 'Казакав', 'Ефiмав', 'Денiсав',
		'Громав', 'Фамiн', 'Давыдав', 'Мельнiкав', 'Шчэрбаков', 'Блiнав', 'Колеснiкав',
		'Карпав', 'Афанасьев', 'Власав', 'Маслав', 'iсакав', 'Цiханав', 'Аксёнав',
		'Гаврiлав', 'Родiонав', 'Котав', 'Горбунав', 'Кудряшов', 'Быкав', 'Зуев',
		'Третьякав', 'Савельев', 'Панав', 'Рыбакав', 'Суворав', 'Абрамав', 'Воронав',
		'Мухiн', 'Архiпав', 'Трофiмав', 'Мартынав', 'Емельянав', 'Гаршков', 'Чарнов',
		'Аучыннiкав', 'Селязнёв', 'Панфiлав', 'Копылав', 'Мiхеев', 'Галкiн', 'Назарав',
		'Лобанав', 'Лукiн', 'Белякав', 'Потапав', 'Некрасав', 'Хохлав', 'Жданав',
		'Наумав', 'Шылав', 'Воронцав', 'Ермаков', 'Драздов', 'iгнатьев', 'Савiн',
		'Логiнав', 'Сафонав', 'Капустiн', 'Кiрiллав', 'Моiсеев', 'Елiсеев', 'Кошелев',
		'Костiн', 'Гарбачов', 'Орехав', 'Ефрэмав', 'Iсаев', 'Еудокiмав', 'Калашнiкав',
		'Кабанав', 'Насков', 'Юдiн', 'Кулагiн', 'Лапiн', 'Прохарав', 'Несцерав',
		'Харiтонав', 'Агафонав', 'Муравьёв', 'Ларыонав', 'Федосеев', 'Зiмiн', 'Пахомав',
		'Шубiн', 'iгнатав', 'Фiлатав', 'Крукав', 'Рогав', 'Кулакав', 'Цярэнцьев',
		'Молчанав', 'Владiмiрав', 'Арцемьев', 'Гурьев', 'Зiнавьев', 'Грiшын', 'Конанав',
		'Дементьев', 'Сiтнiкав', 'Сiмонав', 'Мiшын', 'Фадеев', 'Камiсарав', 'Мамонтав',
		'Носав', 'Гуляев', 'Шарав', 'Устiнав', 'Вiшнякав', 'Евсеев', 'Лаурэнцев',
		'Брагiн', 'Канстанцiнав', 'Корнiлав', 'Аудзеев', 'Зыкав', 'Бiрюков', 'Шарапав',
		'Нiканав', 'Шчукiн', 'Дьячков', 'Адiнцов', 'Сазонав', 'Якушов', 'Красiльнiкав',
		'Гардзеев', 'Самойлав', 'Князев', 'Бяспалав', 'Уварав', 'Шашков', 'Бобылёв',
		'Даронiн', 'Белазёрав', 'Рожкав', 'Самсонав', 'Мяснiкав', 'Лiхачёв', 'Бурав',
		'Сысоев', 'Фамiчов', 'Русакав', 'Стралков', 'Гушчын', 'Цецерiн', 'Колобав',
		'Субботiн', 'Фокiн', 'Блохiн', 'Селiверстав', 'Пестав', 'Кондраццев', 'Сiлiн',
		'Меркушов', 'Лыткiн', 'Турав'
	],

	first_name_male: function() {
		return this.random_element(this.first_name_males);
	},

	first_name_female: function() {
		return this.random_element(this.first_name_females);
	},

	middle_name_male: function() {
		return this.random_element(this.middle_name_males);
	},

	middle_name_female: function() {
		return this.random_element(this.middle_name_females);
	},

	last_name_male: function() {
		return this.random_element(this.last_name_males);
	},

	last_name_female: function() {
		return this.random_element(this.last_name_females)+'a';
	},

	first_name: function() {
		if (this.integer % 2) {
			return this.first_name_male;
		}

		return this.first_name_female;
	},

	last_name: function() {
		if (this.integer % 2) {
			return this.last_name_male;
		}

		return this.last_name_female;
	}
};

module.exports = provider;
